apps/examples/testcase/le_tc/kernel/tc_mqueue.c

```diff
@@ -573,18 +698,114 @@ static void tc_mqueue_mq_unlink(void)
 
 	mq_close(mqdes);
+
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @fn                   :tc_mqueue_mq_timedsend_timedreceive_failurechecks
+* @description          :Function for tc_mqueue_mq_timedsend_timedreceive corner case checks
+* @return               :void*
+*/
+static void tc_mqueue_mq_timedsend_timedreceive_failurechecks(void)
+{
+	char msg_buffer[TEST_MSGLEN];
+	struct mq_attr attr;
+	struct timespec st_time;
+	int ret_chk = OK;
+
+	/* TIMEDSEND_TEST */
+
+	clock_gettime(CLOCK_REALTIME, &st_time);
+	st_time.tv_sec += 1;
+
+	/* Fill in attributes for message queue */
+
+	attr.mq_maxmsg = TEST_TIMEDRECEIVE_NMSGS;
+	attr.mq_msgsize = TEST_MSGLEN;
+	attr.mq_flags = 0;
+
+	/* Fill in a test message buffer to send */
+
+	memcpy(msg_buffer, TEST_MESSAGE, TEST_MSGLEN);
+
+	/* Perform the timedsend_test for NULL time parameter check */
+
+	g_timedsend_mqfd = mq_open("t_mqueue", O_WRONLY | O_CREAT, 0666, &attr);
+	TC_ASSERT_NEQ("timedsend_test", g_timedsend_mqfd, (mqd_t)ERROR);
+
+	/* Perform the send for above failure check case */
+
+	ret_chk = mq_timedsend(NULL, msg_buffer, TEST_MSGLEN, 42, NULL);
+	TC_ASSERT_EQ_CLEANUP("timedsend_test", ret_chk, ERROR, goto cleanup1);
+	TC_ASSERT_EQ_CLEANUP("timedsend_test", errno, EINVAL, goto cleanup1);
+
+	/* TIMEDRECEIVE_TEST */
+
+	clock_gettime(CLOCK_REALTIME, &st_time);
+	st_time.tv_sec += 1;
+
+	/* Fill in attributes for message queue */
+
+	attr.mq_maxmsg = TEST_TIMEDRECEIVE_NMSGS;
+	attr.mq_msgsize = TEST_MSGLEN;
+	attr.mq_flags = 0;
+
+	/* Perform the timedreceive_test for time-related errors */
+
+	memset(msg_buffer, 0xaa, TEST_MSGLEN);
+
+        g_timedrecv_mqfd = mq_open("t_mqueue", O_RDONLY | O_CREAT, 0666, &attr);
+	TC_ASSERT_NEQ_CLEANUP("timedreceive_test", g_timedrecv_mqfd, (mqd_t)ERROR, goto cleanup1);
+
+	/* Perform the receive for above failure check cases */
+
+	ret_chk = mq_timedreceive(g_timedrecv_mqfd, msg_buffer, TEST_MSGLEN, 0, &st_time);
+	TC_ASSERT_EQ_CLEANUP("timedreceive_test", ret_chk, ERROR, goto cleanup2);
+
+	/* Close the queue and return success */
+
+	TC_ASSERT_GEQ_CLEANUP("timedreceive_test", mq_close(g_timedrecv_mqfd), 0, goto cleanup2);
+
+	clock_gettime(CLOCK_REALTIME, &st_time);
+	st_time.tv_sec += 1;
+
+	sleep(10);
+
+	/* Perform the timedreceive_test for expired time & empty message queue cases */
+
+	memset(msg_buffer, 0xaa, TEST_MSGLEN);
+
+        g_timedrecv_mqfd = mq_open("t_mqueue", O_RDONLY | O_CREAT, 0666, &attr);
+	TC_ASSERT_NEQ_CLEANUP("timedreceive_test", g_timedrecv_mqfd, (mqd_t)ERROR, goto cleanup1);
+
+	/* Perform the receive for above failure check cases */
+
+	ret_chk = mq_timedreceive(g_timedrecv_mqfd, msg_buffer, TEST_MSGLEN, 0, &st_time);
+	TC_ASSERT_EQ_CLEANUP("timedreceive_test", ret_chk, ERROR, goto cleanup2);
+
+	/* Close the queue and return success */
+
+	TC_ASSERT_GEQ("timedsend_test", mq_close(g_timedsend_mqfd), 0);
+	TC_ASSERT_GEQ("timedreceive_test", mq_close(g_timedrecv_mqfd), 0);
+
```

Comment:
Don't you need to unlink them?

Reply:
not sure, it asserts if i am unlinking them

Reply:
After using, you should call mq_unlink. Please refer mqueu on kernel_sample.